template/utils/config.py

```diff
@@ -0,0 +1,180 @@
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
+# and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+# the Software.
+
+# THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+# THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+import os
+import torch
+import argparse
+import bittensor as bt
+from loguru import logger
+
+
+def check_config(cls, config: "bt.Config"):
+    r"""Checks/validates the config namespace object."""
+    bt.logging.check_config(config)
+
+    full_path = os.path.expanduser(
+        "{}/{}/{}/netuid{}/{}".format(
+            config.logging.logging_dir,
+            config.wallet.name,
+            config.wallet.hotkey,
+            config.netuid,
+            config.neuron.name,
+        )
+    )
+    print("full path:", full_path)
+    config.neuron.full_path = os.path.expanduser(full_path)
+    if not os.path.exists(config.neuron.full_path):
+        os.makedirs(config.neuron.full_path, exist_ok=True)
+
+    if not config.neuron.dont_save_events:
+        # Add custom event logger for the events.
+        logger.level("EVENTS", no=38, icon="📝")
+        logger.add(
+            config.neuron.full_path + "/" + "completions.log",
+            rotation=config.neuron.events_retention_size,
+            serialize=True,
+            enqueue=True,
+            backtrace=False,
+            diagnose=False,
+            level="EVENTS",
+            format="{time:YYYY-MM-DD at HH:mm:ss} | {level} | {message}",
+        )
+
+
+def add_args(cls, parser):
+    """
+    Adds relevant arguments to the parser for operation.
+    """
+    # Netuid Arg
+    parser.add_argument(
+        "--netuid", type=int, help="Prompting network netuid", default=1
+    )
+
+    parser.add_argument(
+        "--neuron.name",
+        type=str,
+        help="Trials for this miner go in miner.root / (wallet_cold - wallet_hot) / miner.name. ",
+        default="core_prompting_validator",
+    )
+    parser.add_argument(
+        "--neuron.device",
+        type=str,
+        help="Device to run the validator on.",
+        default="cpu",
+    )
+    parser.add_argument(
+        "--neuron.disable_log_rewards",
+        action="store_true",
+        help="Disable all reward logging, suppresses reward functions and their values from being logged to wandb.",
```

Comment:
I am not sure if we want this in the template, wandb seems to be more specific to subnet 1